src/blight/actions/bitcode_extract.py

```diff
@@ -0,0 +1,51 @@
+"""
+The `BitcodeExtract` action.
+"""
+
+import logging
+import os
+import subprocess
+from typing import List
+
+from blight.action import CompilerAction
+from blight.enums import Lang
+from blight.tool import CompilerTool
+
+logger = logging.getLogger(__name__)
+
+
+class BitcodeExtract(CompilerAction):
+    """
+    Compile and extract bitcode. The output bitcode file will be located in the same direcory
+    as the output executable.
+    """
+
+    def before_run(self, tool: CompilerTool) -> None:  # type: ignore
+        store = self._config.get("store")
+        if store is not None:
```

Comment:
Nit: an early-return style would make this a little easier to read:

```python
if store is None:
    return
    
# ...    
```

Same for throughout this method body.

Reply:
(Alternatively, make this an error condition, since this action doesn't make any sense without a `store`.)